include/seqan3/alignment/pairwise/alignment_configurator.hpp

```diff
@@ -106,8 +106,8 @@ struct alignment_contract
                                     decltype(get<align_cfg::scoring>(std::declval<alignment_config_type>()).value)
                                  >;
             return static_cast<bool>(scoring_scheme<scoring_type,
-                                                   value_type_t<first_seq_t>,
-                                                   value_type_t<second_seq_t>>);
+                                                   std::iter_value_t<first_seq_t>,
+                                                   std::iter_value_t<second_seq_t>>);
```

Comment:
This is the cause of your failing test case, because both sequences are ranges and not iterators.
```suggestion
                                                   std::range::range_value_t<first_seq_t>,
                                                   std::range::range_value_t<second_seq_t>>);
```